Library/Homebrew/extend/os/linux/development_tools.rb

```diff
@@ -8,7 +8,10 @@ class << self
     sig { params(tool: String).returns(T.nilable(Pathname)) }
     def locate(tool)
       (@locate ||= {}).fetch(tool) do |key|
-        @locate[key] = if (path = HOMEBREW_PREFIX/"bin/#{tool}").executable?
+        @locate[key] = if (build_system_too_old? &&
+                            ((path = HOMEBREW_PREFIX/"opt/binutils/bin/#{tool}").executable? ||
+                             (path = HOMEBREW_PREFIX/"opt/glibc/bin/#{tool}").executable?)) ||
+                          (path = HOMEBREW_PREFIX/"bin/#{tool}").executable?
```

Comment:
This is really confusing. I suggest extracting some of this into variables before calling `if`.

Or, alternatively, avoid using `path` as the assigned variable so the linter doesn't complain about duplicated bodies.

Reply:
> Or, alternatively, avoid using `path` as the assigned variable so the linter doesn't complain about duplicated bodies.

That's a good idea.